Spam and Phishing modules added

hiiimken opened the pull request:
- created modules folder
- added spamtest, phishingtest python files, and phising_domain and phishing_links text files
- refactored spamtest and phishingtest, as they were initially coded to be run as a separate program
- refactored spamtest as spam keywords were hardcoded prior. Created a separate text file spam_keywords to be updated if other keywords arise.
- Added both phishing_domains and phishing_links text files to modules folder

hiiimken committed:
Spam and Phishing modules added

- created modules folder
- added spamtest, phishingtest python files, and phising_domain and phishing_links text files
- refactored spamtest and phishingtest, as they were initially coded to be run as a separate program
- refactored spamtest as spam keywords were hardcoded prior. Created a separate text file spam_keywords to be updated if other keywords arise.
- Added both phishing_domains and phishing_links text files to modules folder

tamaraallos commented on .idea/workspace.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Delete this file.

tamaraallos commented on Modules/phishing_domains.txt:
put all your .txt files in a seperate folder with a meaningful name. E.g 'detection_data'.

tamaraallos commented on Modules/phishingtest.py:
Put your script files (.py files) in a seperate folder. Give it a more meaninfgul name like 'detection scripts' or something. 

When you change file/folder structure make sure to reflect it accurately when you import/read the files.

tamaraallos commented on Modules/phishingtest.py:
```diff
@@ -0,0 +1,86 @@
+    Returns:
+        bool: True if the email is phishing, False otherwise.
+    """
+    sender_domain = extract_domain(parseaddr(email_message.get('From'))[1])
```
the file will fail. where is extract_domain function

hiiimken committed:
Spam and Phishing modules v2

- Restored extract_domain function
- Fixed comments within both programs
- Removed keyword "free" within the spam_keywords text file, as this keywords brings false positives.
- Changed implementation of is_spam on spamtest.py. Instead of iterating through the entire message, implemented it so that it instantly returns the result if it detects a keyword.
- Added error checking with the load_phishing_domains and load_phishing_links function
- Removed unused imports from both files
- Moved files into two separate folders based on usage: spam filter and phishing filter.
